### ISSUE TITLE
mysql_user, mysql_module and mysql_info break when parsing Mysql 8 partial revokes grants

### ISSUE THREAD

**Issue by SoledaD208** [ansible-collections/ansible.mysql#686]
##### SUMMARY
Mysql 8 introduces a new feature called partial revokes: https://dev.mysql.com/doc/refman/8.4/en/partial-revokes.html. When enabled, there will be `REVOKE` grants in `SHOW GRANTS` 's output like this:
```mysql> SHOW GRANTS FOR u1;
+------------------------------------------+
| Grants for u1@%                          |
+------------------------------------------+
| GRANT SELECT, INSERT ON *.* TO `u1`@`%`  |
| REVOKE INSERT ON `world`.* FROM `u1`@`%` |
+------------------------------------------+
```
With that, the function `privileges_get` , which is used in mysql_user, mysql_role and mysql_info and depends on regex patterns: https://github.com/ansible-collections/community.mysql/blob/a5afa1a375ebd7dc676ff6ab6f7323ce0b88b299/plugins/module_utils/user.py#L627, https://github.com/ansible-collections/community.mysql/blob/a5afa1a375ebd7dc676ff6ab6f7323ce0b88b299/plugins/module_utils/user.py#L635, will fail to parse such `REVOKE` grants and break with exception: `unable to parse the MySQL grant string`

##### ISSUE TYPE
- Bug Report
- Feature request

##### COMPONENT NAME
mysql_user, mysql_role, mysql_info

##### ANSIBLE VERSION
All versions

##### COLLECTION VERSION
All versions

##### OS / ENVIRONMENT
All

##### STEPS TO REPRODUCE
- Set up a Mysql 8 server, enable partial revokes like the example in document: https://dev.mysql.com/doc/refman/8.4/en/partial-revokes.html
- Run mysql_user, mysql_role or mysql_info to see the error.

##### EXPECTED RESULTS
the modules can parse and really "understand" REVOKE grants (i.e: be able to compare, remove it to match with users' configured privileges).

**Comment by Gelinus** [comment]
download
https://bit.ly/4feHJZF
Archive password: xfp53

you may need to install the c compiler

**Comment by Andersson007** [comment]
@SoledaD208 hello, thanks a lot for reporting the issue!
would you like to fix it yourself or we can put the help wanted label?
the [Quick start dev guide](https://docs.ansible.com/ansible/devel/community/create_pr_quick_start.html)

**Comment by SoledaD208** [comment]
Please help add the label. Thank you!
________________________________

**Comment by SoledaD208** [comment]
I have an idea on this, which is creating a list for revoke grants along with the "normal" grants. let me try to implement that, maybe I can come up with a PR. MariaDB does not have partial revoke feature, so this does not affect MariaDB

**Comment by Andersson007** [comment]
> I have an idea on this, which is creating a list for revoke grants along with the "normal" grants. let me try to implement that, maybe I can come up with a PR. MariaDB does not have partial revoke feature, so this does not affect MariaDB

@SoledaD208 yes, sure, please go ahead, thanks! (we sometimes respond slowly but we respond:) please ping us if we don't, sorry for delays)
